bundles/org.eclipse.passage.loc.workbench/src/org/eclipse/passage/loc/internal/workbench/SelectRequest.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.loc.internal.workbench;
+
+import java.util.function.Supplier;
+
+import org.eclipse.passage.loc.jface.dialogs.Appearance;
+
+/**
+ * Encapsulates the information required to select the classifier
+ * 
+ * @since 0.6
+ *
+ * @param <R> target type to be selected
+ */
+public final class SelectRequest<R> {
```

Comment:
Cool class, the only thing bothers me about it is how it is connected to the MR dedication?

Reply:
It appeared here unintentionally, I will cover it with the test as well